tests/ui/mut_key.rs

```diff
@@ -52,4 +61,16 @@ fn main() {
     tuples::<Key>(&mut HashMap::new());
     tuples::<()>(&mut HashMap::new());
     tuples_bad::<()>(&mut HashMap::new());
+
+    raw_ptr_is_ok(&mut HashMap::new());
+    raw_mut_ptr_is_ok(&mut HashMap::new());
+
+    // Currently fail as expected:
+    let _map = HashMap::<Cell<usize>, usize>::new();
+    let _map = HashMap::<&mut Cell<usize>, usize>::new();
+    // Collection types from `std` which hash/order their type parameter
```

Comment:
not sure what you mean by "order"

Reply:
This lint triggers on both `Hash(Map|Set)` (which requires `T: Hash`) as well as `BTree(Map|Set)` (which requires `T: Ord`).  Interior mutability is bad in both cases, but 'order' was referring to `Ord::cmp` required by `BTree(Map|Set)`

Reply:
Right I've updated the comment.